pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -145,10 +145,37 @@ type ReconcileKieApp struct {
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (reconciler *ReconcileKieApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	logrus.Debugf("Reconciling %s/%s\n", request.Namespace, request.Name)
+	logrus.Debugf("Reconciling %s/%s", request.Namespace, request.Name)
 
-	// Fetch/Create critical ConfigMaps
-	reconciler.initConfigMaps(request.Namespace)
+	// Create critical ConfigMaps if don't exist
+	configMaps := defaults.ConfigMapsFromFile(request.Namespace)
+	for _, configMap := range configMaps {
+		var testDir bool
+		result := strings.Split(configMap.Name, "-")
+		if len(result) > 1 {
+			if result[1] == "testdata" {
```

Comment:
It's not clear to me that this testdata directory will be a viable testing strategy for us. However, it looks like you're also looking at the client object to determine it's a unit test, so it should work for tests relying on our main yaml files too, right?

Reply:
correct, works for both. this piece of code just keeps the operator from generating configmaps from these test files.